lib/imagegallery-block.rb

```diff
@@ -0,0 +1,67 @@
+require 'asciidoctor'
+require 'asciidoctor/extensions'
+counter = 'a'
+
+Asciidoctor::Extensions.register do
+  block do
+    named :gallery
+    on_context :open
+    # A bit hacky. Each image gallery needs a unique string:
+    # http://stackoverflow.com/questions/88311/how-best-to-generate-a-random-string-in-ruby#88341
+    o = [('a'..'z'), ('A'..'Z')].map { |i| i.to_a }.flatten
+    process do |parent, reader, attrs|
+      lines = []
+      wrapper = create_open_block parent, [], {}
+      parse_content wrapper, reader
+      counter = (0...50).map { o[rand(o.length)] }.join
+      wrapper.find_by(context: :image).each do |img|
+        src = img.attr 'target'
+        alt = img.attr 'alt'
+        thumb_src = src.sub(/\.[^.]+$/, '-thumb\0')
+        #lines << %(image:#{thumb_src}[#{alt}, title="#{img.attr 'title'}", role=fancybox-effects, link=#{parent.image_uri src}])
+        lines << %(<a class="fancybox" rel="fancybox-#{counter}" href="#{parent.image_uri src}" title="#{img.attr 'title'}"><img src="#{parent.image_uri thumb_src}" alt="#{alt}"></a>)
+      end
+      create_paragraph parent, lines, {}, subs: nil
+    end
+  end
+
+  docinfo_processor do
+    at_location :head
+    process do |doc|
+      result = []
+      backend = doc.backend
+      if backend == 'html5'
+        result << %(
```

Comment:
The `result` Array is not needed here. You can just remove it and replace
```
result << %(
```
with
```
%(
```